pkg/runtime/handlerTable.go

```diff
@@ -145,20 +146,33 @@ func (t *handlerTable) initHandler(he *HandlerEntry) {
 	he.Handler, he.HandlerCreateError = t.buildHandler(hc, insts)
 }
 
-func encode(enc *gob.Encoder, e interface{}) {
-	if err := enc.Encode(e); err != nil {
-		glog.Warningf("Unable to encode %v", e)
+func encode(buf *bytes.Buffer, v interface{}) {
+	switch t := v.(type) {
+	case string:
+		if _, err := buf.Write([]byte(t)); err != nil {
+			glog.Warningf("Failed to write %v to a buffer: %v", t, err)
+		}
+	case proto.Message:
+		if b, err := proto.Marshal(proto.Message(t)); err != nil {
+			glog.Warningf("Failed to marshall %v into a proto: %v", t, err)
+		} else if _, err := buf.Write(b); err != nil {
+			glog.Warningf("Failed to write %v to buffer: %v", b, err)
+		}
+	default:
+		glog.Warningf("Fell into default case for v.(type): %#v; falling back to fmt.Fprintf()", t)
+		if _, err := fmt.Fprintf(buf, "%v", t); err != nil {
+			glog.Warningf("Failed to write %v to buffer: %v", t, err)
+		}
 	}
 }
 
 // computeSha for individual handler entries
 func (t *handlerTable) computeSha() {
-	var buff bytes.Buffer
-	enc := gob.NewEncoder(&buff)
+	buf := new(bytes.Buffer)
```

Comment:
`var buf bytes.Buffer`  is preferred.

Reply:
We need a pointer to a buffer, not the buffer itself (`*bytes.Buffer` implements `io.Writer`, `bytes.Buffer` does not). `new(bytes.Buffer)` and `&bytes.Buffer{}` are identical.

Reply:
ok.